### PR TITLE
Add persisting of created charge link

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## Description

Add persisting of (domain) created charge link.
Uses EF Core fluent configuration.

## References

* #334

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Context/ChargesDatabaseContext.cs [hunk 1]
```diff
@@ -64,13 +64,58 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             modelBuilder.Entity<ChargeLink>(builder =>
```
Comment (Mech0z) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Context/ChargesDatabaseContext.cs:64]: Should we move the configuration into separate files like https://github.com/Energinet-DataHub/geh-metering-point/blob/main/source/Energinet.DataHub.MeteringPoints.Infrastructure/DataAccess/MeteringPoints/MeteringPointEntityConfiguration.cs

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Context/ChargesDatabaseContext.cs [hunk 2]
```diff
@@ -64,13 +64,58 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
...
+
+                        details.HasKey(p => p.Id);
+
+                        details.Property(p => p.Id).ValueGeneratedNever();
+
+                        details.Property(d => d.Factor).HasColumnName("Factor");
+
+                        details.Property(d => d.CreatedByOperationId).HasColumnName("CreatedByOperationId");
+
+                        details.Property(d => d.RetiredByOperationId).HasColumnName("RetiredByOperationId");
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Context/ChargesDatabaseContext.cs:109]: Is _HasColumnName_ necessary when property and column name are equal?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Infrastructure/Repositories/ChargeLinkRepositoryTests.cs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+using Xunit;
+using Xunit.Categories;
+using MarketParticipant = GreenEnergyHub.Charges.Infrastructure.Context.Model.MarketParticipant;
+using MeteringPoint = GreenEnergyHub.Charges.Infrastructure.Context.Model.MeteringPoint;
+
+namespace GreenEnergyHub.Charges.Tests.Infrastructure.Repositories
+{
+    /// <summary>
+    /// Tests <see cref="ChargeRepository"/> using an SQLite in-memory database.
+    /// </summary>
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Infrastructure/Repositories/ChargeLinkRepositoryTests.cs:36]: Still using SQLite?

**Review by HenrikSommer — APPROVED** [review]
Left a few comments, but generally looks good to me

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Context/ChargesDatabaseContext.cs (hunk 1 above)
Comment (BjarkeMeier) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Context/ChargesDatabaseContext.cs]: Certainly :-) Done.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Context/ChargesDatabaseContext.cs (hunk 2 above)
Comment (BjarkeMeier) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Context/ChargesDatabaseContext.cs]: Very good question. No, it's not needed. I, however, thought about it decided to be explicit about the naming everywhere. I prefer that and it's in accordance with how it's done in the metering point domain.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Infrastructure/Repositories/ChargeLinkRepositoryTests.cs (hunk 3 above)
Comment (BjarkeMeier) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Infrastructure/Repositories/ChargeLinkRepositoryTests.cs]: Thx. Removed.

**Review by Mech0z — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Test/202109080933 Metering Point Test Data.sql [hunk 4]
```diff
@@ -0,0 +1,2 @@
+﻿INSERT INTO [Charges].[MeteringPoint] VALUES ('571313180000000005', 1, 800, '2021-01-01', 1, 3)
```
Comment (bemwen) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Test/202109080933 Metering Point Test Data.sql:1]: The effective date will probably come back to haunt us later unless we use UTC on a date switch. For example, 2020-12-31 23:00'

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202109071000 Refactor ChargeLink Model.sql [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+CREATE INDEX IX_ChargeLinkId ON [Charges].[ChargeLinkOperation] (ChargeLinkId DESC, WriteDateTime DESC);
+GO
+
+CREATE TABLE [Charges].[ChargeLinkPeriodDetails]
+(
+    Id UNIQUEIDENTIFIER NOT NULL,
+    ChargeLinkId UNIQUEIDENTIFIER NOT NULL FOREIGN KEY REFERENCES [Charges].ChargeLink(Id),
+    StartDateTime DateTime2 NOT NULL,
+    EndDateTime DateTime2,
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202109071000 Refactor ChargeLink Model.sql:42]: As we became aware of today, let's make this: 
```suggestion
    EndDateTime DateTime2 NOT NULL,
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/ChargeLinks/ChargeLinkOperation.cs [hunk 6]
```diff
@@ -22,51 +22,46 @@ public class ChargeLinkOperation
...
         }
 
         /// <summary>
         /// Globally unique identifier of the charge link operation.
         /// </summary>
-        public int? RowId { get; set; }
+        public Guid Id { get; }
 
         /// <summary>
         /// Operation ID provided by customer.
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/ChargeLinks/ChargeLinkOperation.cs:55]:
```suggestion
        /// Contains an ID for the specific Charge Link Operation, provided by the sender. Uniqueness cannot be guaranteed.
```

**Review by prtandrup — APPROVED** [review]
LGTM - CustomerProvidedId to SenderProvidedId and I'm happy to 🛩️

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Test/202109080933 Metering Point Test Data.sql (hunk 4 above)
Comment (BjarkeMeier) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Test/202109080933 Metering Point Test Data.sql]: Thx. Fixed.

**Pull request merged**